src/q7/crypto-MAT2450/exercises/crypto-MAT2450-exercises.tex

```diff
@@ -19,37 +23,84 @@
 \DeclareMathOperator{\Open}{\mathsf{Open}}
 \DeclareMathOperator{\Sign}{\mathsf{Sign}}
 \DeclareMathOperator{\Sig}{\mathsf{Sig}}
+\DeclareMathOperator{\PRG}{\mathsf{PRG}}
+\DeclareMathOperator{\F}{\mathsf{F}}
+\DeclareMathOperator{\f}{\mathsf{f}}
+\DeclareMathOperator{\sprp}{\mathsf{sprp}}
+\DeclareMathOperator{\adv}{\mathsf{adv}}
+\DeclareMathOperator{\Gr}{\mathsf{Gr}}
+\DeclareMathOperator{\Ima}{\mathsf{Im}}
+
+
+% Experiments
+\newcommand{\PrivKeav}{\mathsf{PrivK}_{\A, \Pi}^{\mathsf{eav}}}
+\newcommand{\PrivKmult}{\mathsf{PrivK}_{\A,\Pi}^\mathsf{mult}}
+\newcommand{\PrivKcpa}{\mathsf{PrivK}_{\A,\Pi}^\mathsf{cpa}}
+\newcommand{\PrivKcca}{\mathsf{PrivK}_{\A,\Pi}^\mathsf{cca}}
+
+% Functions
+\newcommand{\negl}{\ensuremath{\epsilon}}
 
+% Adversary
+\newcommand{\A}{\mathcal{A}}
+\newcommand{\Adv}{\ensuremath{\mathcal{A}}}
+
+% Arithmetic
+\newcommand{\modn}{\, (\text{mod } n)}
+
+
+\newcommand{\xor}{\oplus}
+\newcommand{\st}{\text{ s.t. }}
+\renewcommand{\Z}{\mathbb{Z}}
 \DeclareMathOperator{\PrivK}{PrivK}
 \DeclareMathOperator{\MacForge}{MacForge}
 \DeclareMathOperator{\Sigforge}{Sig-forge}
 \DeclareMathOperator{\Invert}{Invert}
 \DeclareMathOperator{\DDH}{DDH}
 \DeclareMathOperator{\DLog}{DLog}
-\newcommand{\PrivKeav}{\PrivK^{\text{eav}}}
 \newcommand{\PrivKmultcpa}{\PrivK^{\text{multcpa}}}
-\newcommand{\Sigforgeone}{\Sigforge^{\text{1--time}}}
+\newcommand{\Sigforgeone}{\Sigforge^{\text{1-time}}}
 
-\DeclareMathOperator{\Ima}{\mathsf{Im}}
+% New commands
+\newcommand{\flec}{\draw[->,>=latex]}
+\newcommand{\flect}{\draw[->,>=latex,red]}
+\newcommand{\flecc}{\draw[->,>=latex,blue]}
+\newcommand{\mess}{node[midway,above]}
+\newcommand{\messd}{node[midway,below]}
+\newcommand{\trapeze}[2]{\draw #2 node {#1};
+	\draw[line width=1pt] #2 ++(-1,2) -- ++(2,-1) {[rounded corners] -- ++(0,-2) -- ++(-2,-0)} -- ++(0,3);}
+\newcommand{\position}{(0,0)}
+\newcommand{\mylabel}{$x_1$}
+\newcommand{\carref}[2] { \draw[line width = 1pt,fill=#2] #1 ++ (-1,-1) rectangle #1 ++ (1,1); }
+\newcommand{\carrep}[3] { \draw[line width = 1pt,pattern color=#2,pattern= #3] #1 ++ (-1,-1) rectangle #1 ++ (1,1); }
 
-\hypertitle{Cryptography}{7}{MAT}{2450}
-{Benoît Legat}
-{François Koeune, Olivier Pereira}
-
-\include{APE1}
-
-\include{APE2}
 
-\include{APE3}
+\titleformat
+{\subsection}[hang]{\bfseries\Large}{\thesubsection: }{0.5ex}{}
 
-\include{APE4}
-
-\include{APE5}
+\hypertitle{Cryptography}{7}{MAT}{2450}
+{Benoît Legat and Luis Tascon Gutierrez and Guillaume Gheysen and Olivier Leblanc}
```

Comment:
Je suis désolé, mais j'avais juste demandé de mettre `and` pour les professeurs, pas les auteurs.
Il faut:
* \and pour les auteurs, car ça crée un espace entre ceux-ci
* and (en EN) ou et (en FR) pour les professeurs car c'est inséré tel quel dans le texte de la 1e page

Aussi, je pense que tu te compliques la tâche en faisant ces petites modifs en local. L'utilité du "suggested changes" que je fais à chaque fois, c'est que tu peux faire ces changements directement sur GitHub en un rien de temps. Donc je te conseille de juste cliquer sur "commit suggestion" ci-dessous, c'est la seule chose à faire. :wink: 
```suggestion
{Benoît Legat \and Luis Tascon Gutierrez \and Guillaume Gheysen \and Olivier Leblanc}
```

Reply:
Je ne suis pas d'accord avec toi Martin : le soucis de ces "suggested changes" c'est que c'est commité sur GitHub mais pas en local, donc si l'auteur a envie de faire par la suite des modifications en local (par exemple faire une grosse modif, réarranger les commits ou autre), il doit pull des changements depuis GitHub, alors que c'est sa propre branche (=awkward).

Reply:
Merci, mais maintenant, git a ajouté des lignes quand tu as pull ta branche (cause de conflit).
Donc il faut que tu supprimes ces lignes (on voit un <<<<<<<<<< HEAD ).

Reply:
@Jimvy et @MartinBraquet d'autres commentaires ou on est bon?

Reply:
Yes :ok_hand: